### PR TITLE
Allow Symfony 6

### PR DESCRIPTION
**Pull request by dbu**
continue #821

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            - name: Checkout project
+              uses: actions/checkout@v2
+
+            - name: Install and configure PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  tools: 'composer:v2'
+
+            - name: Globally install symfony/flex
```
Comment (acrobat) [inline .github/workflows/test-application.yaml:70]: You could replace this by adding it to the tools section of `shivammathur/setup-php` 

```yaml
        - name: Install and configure PHP
              uses: shivammathur/setup-php@v2
              with:
                  php-version: ${{ matrix.php-version }}
                  tools: 'composer:v2, flex'
```

File: .github/workflows/test-application.yaml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+              if: matrix.symfony-require != ''
+              run: composer global require --no-progress --no-scripts --no-plugins symfony/flex
+
+            - name: Remove dev dependencies not compatible with Symfony 6
```
Comment (acrobat) [inline .github/workflows/test-application.yaml:74]: Should be ok with 1.6.4 release, https://github.com/liip/RMT/commit/f14c83070565cd066d27648c75d6448a03a8b825

File: tests/Doctrine/Tests/ODM/PHPCR/Mapping/AbstractMappingDriverTest.php [hunk 3]
```diff
@@ -396,7 +412,7 @@ public function testLoadChildrenMapping()
      */
     public function testChildrenMapping(ClassMetadata $class)
     {
-        $this->assertInternalType('array', $class->childrenMappings);
+        $this->assertIsArray($class->childrenMappings);
         $this->assertCount(2, $class->childrenMappings);
         $this->assertArrayHasKey('all', $class->mappings);
         $this->assertArrayNotHasKey('filter', $class->mappings['all']);
```
Comment (acrobat) [inline tests/Doctrine/Tests/ODM/PHPCR/Mapping/AbstractMappingDriverTest.php:418]: I was checking the tests locally to help move this PR forward, but the failing test here is actually a bug introduces some time ago. 

```diff
- $this->assertFalse(isset($class->mappings['all']['filter']));
+ $this->assertArrayNotHasKey('filter', $class->mappings['all']);
```

But the filter value in the array is `null` in this case. So the previous isset check would work but the current check fails. This needs to be changed to `$this->assertNull($class->mappings['all']['filter']);`

See: https://github.com/doctrine/phpcr-odm/commit/3f23224b00b07097fb5da5e62f2b5bcb5b0fe8be#diff-87c636fa83a2393c03d9d0cbc6c2b33c90cf07e9c0ce08f8c990872df4588f84R395-R402

**Comment by acrobat** [comment]
I've created #823 with fixes to make the tests pass again. See https://github.com/acrobat/phpcr-odm/runs/4431734943?check_suite_focus=true for the test run

File: tests/Doctrine/Tests/ODM/PHPCR/Mapping/AbstractMappingDriverTest.php (hunk 3 above)
Comment (dbu) [reply tests/Doctrine/Tests/ODM/PHPCR/Mapping/AbstractMappingDriverTest.php:418]: thanks, that explains it. and goes to show that merging without the ci working is dangerous.

File: .github/workflows/test-application.yaml (hunk 1 above)
Comment (dbu) [reply .github/workflows/test-application.yaml]: thanks!

File: .github/workflows/test-application.yaml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                    - '7.2'
+                    - '7.3'
+                    - '7.4'
+                    - '8.0'
+                dependencies: [highest]
+                symfony-version: ['*']
+                include:
+                    - php-version: '7.1'
+                      dependencies: lowest
+                      symfony-version: *
```
Comment (acrobat) [inline .github/workflows/test-application.yaml:40]:
```suggestion
                      symfony-version: '*'
```

File: .github/workflows/test-application.yaml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
```
Comment (acrobat) [inline .github/workflows/test-application.yaml:117]:
```suggestion
                      symfony-version: '*'
```

**Comment by acrobat** [comment]
To fix the remaining errors about the uuid field, this patch should be applied.

```diff
diff --git a/tests/Doctrine/Tests/ODM/PHPCR/Mapping/Model/UuidMappingObject.php b/tests/Doctrine/Tests/ODM/PHPCR/Mapping/Model/UuidMappingObject.php
index 17980035..e8ce8abe 100644
--- a/tests/Doctrine/Tests/ODM/PHPCR/Mapping/Model/UuidMappingObject.php
+++ b/tests/Doctrine/Tests/ODM/PHPCR/Mapping/Model/UuidMappingObject.php
@@ -12,6 +12,6 @@ class UuidMappingObject
     /** @PHPCRODM\Id */
     public $id;
 
-    /**@PHPCRODM\Uuid()*/
+    /** @PHPCRODM\Uuid() */
     public $uuid;
 }
```

**Comment by dbu** [comment]
alright, this is coming together. thanks a lot for the work @jordisala1991 and @acrobat !

i realized that `master` is whats to become version 2 eventually. i rebased this onto the 1.x branch so that i can tag a release right away, see #824

(and once #824 is merged, i will update master from 1.x)

**Pull request closed without merging**